### ISSUE TITLE
ClassClock time does not match between two devices

### ISSUE THREAD

**Issue by MoralCode** [MoralCode/ClassClock#21]
When comparing ClassClock's countdown between two devices, the times may not match. This is due to the fact that ClassClock uses the time of the device it is loaded on (using javascript's `Date()` function) and inconsistencies between the time settings of different devices may cause differences in the displayed time. 

# Temporary workaround:

**Comment by MoralCode** [comment]
another fix since many schools may already be using a wireless/synchronized clock system (like https://www.innovationwireless.com/markets/k12-schools.html), ClassClock may be able to find a way to hook into that.

**Comment by polygonnedpotato** [comment]
It could be possible to connect to a time server such as time.nist.gov or time.windows.com

**Comment by MoralCode** [comment]
yeah i was thinking about that, although i think those specific services operate using the NTP protocol which may be a little overkill for this use case as it only really needs to be accurate to within a second or two. 

The other thing i was thinking of was to just use the time of the server where the ClassClock web app is being hosted, which should be able to just be read from the `Date` header of the responses from the API.

Having a "time source" that can be set on a per-school basis might be better to allow schools to sync ClassClock with whatever their wall clocks are synced from so that things stay consistent with the clocks that students are used to using. However, this per-school time source is a pretty "stretch" feature that i haven't looked too far into.

Because of the issues in #23 with the `development` branch being kinda disorganized and blocked by the admin panel addition and no free time on my side to work on it, I haven't gotten around to implementing any kind of time-syncing yet.

**Comment by MoralCode** [comment]
Since this issue has been opened i have also been working on improving the accuracy of https://github.com/NodeGuy/server-date so all the classclock webapps an just sync their time with the server, so that they are all relatively close to each other time-wise (id say within 100 or 200 ms is probably acceptable for this. sub-100 would be nice if it is not too much additional effort though (as thats the limit Ive heard of for user perceptibility)

Just wanted to drop this info here mostly for posterity in case someone else decides to pick up this torch and continue it
